index.js

```diff
@@ -62,6 +63,8 @@ module.exports = function (opts) {
 				}
 
 				if (stat.isDirectory() || entry.fileName.charAt(entry.fileName.length - 1) === '/') {
+					new Mode(stat).isDirectory(true);
```

Comment:
```js
var Decompress = require('decompress');
var unzip = require('decompress-unzip');

new Decompress()
    .src('master.zip')
    .use(unzip({ strip: 1 }))
    .run(function (err, files) {
        files.forEach(function (file) {
            console.log(file.stat.mode, file.path);
        });
    });
```

```
undefined '.gitignore'
undefined 'markup'
undefined 'README.md'
undefined 'markup/modules'
undefined 'markup/modules/_template'
33261 'markup/modules/_template/_template.html'
undefined 'markup/modules/footer'
undefined 'markup/modules/head'
33261 'markup/modules/footer/footer.html'
undefined 'markup/modules/head/data'
undefined 'markup/modules/head/data/data.js'
undefined 'markup/pages'
33261 'markup/modules/head/head.html'
33261 'markup/pages/_template.html'
undefined 'tars'
33261 'markup/pages/index.html'
undefined 'tars/tasks'
33261 'tars/tasks/compile-templates.js'
```

It sometimes fails (`README.md` etc.) but sometime succeeds (`markup/pages/_template.html` etc.).